cpp/src/arrow/CMakeLists.txt

```diff
@@ -80,7 +80,7 @@ if(ARROW_WITH_ZSTD)
 endif()
 
 if(ARROW_ORC)
-  if(ORC_SOURCE STREQUAL "SYSTEM")
+  if(orc_SOURCE STREQUAL "SYSTEM")
```

Comment:
Not strictly related to this change but while taking a look on where are we defining `orc_SOURCE` here:
https://github.com/apache/arrow/blob/8dd9eba99a69dbdf1d110cebf0bd349ee3ac7eda/cpp/cmake_modules/ThirdpartyToolchain.cmake#L91-L97

I came up with:
https://github.com/apache/arrow/blob/8dd9eba99a69dbdf1d110cebf0bd349ee3ac7eda/cpp/cmake_modules/ThirdpartyToolchain.cmake#L113-L119

Shouldn't we do the same below changing `GLOG_SOURCE` for `glog_SOURCE`?
https://github.com/apache/arrow/blob/8dd9eba99a69dbdf1d110cebf0bd349ee3ac7eda/cpp/src/arrow/CMakeLists.txt#L89-L91